src/betterproto/__init__.py

```diff
@@ -455,20 +455,19 @@ def _dump_float(value: float) -> Union[float, str]:
 
     Parameters
     ----------
-    value : float
+    value: float
         Value to dump
 
     Returns
     -------
     Union[float, str]
-        Dumped valid, either a float or the strings
-        "Infinity" or "-Infinity"
+        Dumped value, either a float or the strings
     """
     if value == float("inf"):
         return INFINITY
     if value == -float("inf"):
         return NEG_INFINITY
-    if value == float("nan"):
+    if isinstance(value, float) and math.isnan(value):
```

Comment:
@Gobot1234 should we skip the other checks as well if the instance is not a float? Seems pointless if it is not a float.

Reply:
I think a guard against None in to/from_dict would be the best solution

Reply:
I'll do this in #367 just cause I want to get this merged